unlock-js/src/__tests__/v02/createLock.test.js

```diff
@@ -60,30 +62,76 @@ describe('v02', () => {
       setupFail = fail
     }
 
-    it('should invoke _handleMethodCall with the right params', async () => {
-      expect.assertions(2)
+    describe('when not explicitly providing the address of a denominating currency contract ', () => {
+      it('should invoke _handleMethodCall with the right params', async () => {
+        expect.assertions(2)
 
-      await nockBeforeEach()
-      setupSuccess()
+        await nockBeforeEach()
+        setupSuccess()
 
-      walletService._handleMethodCall = jest.fn(() =>
-        Promise.resolve(transaction.hash)
-      )
-      const mock = walletService._handleMethodCall
+        walletService._handleMethodCall = jest.fn(() =>
+          Promise.resolve(transaction.hash)
+        )
+        const mock = walletService._handleMethodCall
 
-      await walletService.createLock(lock, owner)
+        await walletService.createLock(lock, owner)
 
-      expect(mock).toHaveBeenCalledWith(
-        expect.any(Promise),
-        TransactionTypes.LOCK_CREATION
-      )
+        expect(mock).toHaveBeenCalledWith(
+          expect.any(Promise),
+          TransactionTypes.LOCK_CREATION
+        )
 
-      // verify that the promise passed to _handleMethodCall actually resolves
-      // to the result the chain returns from a sendTransaction call to createLock
-      const result = await mock.mock.calls[0][0]
-      await result.wait()
-      expect(result).toEqual(transactionResult)
-      await nock.resolveWhenAllNocksUsed()
+        // verify that the promise passed to _handleMethodCall actually resolves
+        // to the result the chain returns from a sendTransaction call to createLock
+        const result = await mock.mock.calls[0][0]
+        await result.wait()
+        expect(result).toEqual(transactionResult)
+        await nock.resolveWhenAllNocksUsed()
+      })
+    })
+
+    describe('when providing the address of a denominating currency contract', () => {
+      it('should invoke _handleMethodCall with the right params', async () => {
+        expect.assertions(2)
+
+        await nockBeforeEach()
+
+        let {
+          testTransaction,
+          testTransactionResult,
+          success,
+        } = callMethodData(
+          lock.expirationDuration,
+          testERC20ContractAddress,
+          utils.toWei(lock.keyPrice, 'ether'),
+          lock.maxNumberOfKeys
+        )
+
+        transaction = testTransaction
+        transactionResult = testTransactionResult
+        setupSuccess = success
+
+        setupSuccess()
+
+        walletService._handleMethodCall = jest.fn(() =>
```

Comment:
this test should not mock `_handleMethodCall`, just remove the mock and then the code below